src/tools/BaseMoveTool.ts

```diff
@@ -0,0 +1,58 @@
+import { GlobalStore } from "@/store";
+import BaseTool from "./BaseTool";
+
+/**
+ * BassMoveTool handles the basics of panning the field when meta/alt key is pressed.  Derived tools implement onMouseDown/Up/Move to do tool specific actions.
+ */
+export default abstract class BaseMoveTool {
+  /**
+   * Approximates coordinate on the two step grid
+   *
+   * @param coordinate - Either the x or y coordinate
+   */
+  static enablePan(enable: boolean): void {
+    const grapherSvgPanZoom: SvgPanZoom.Instance | undefined =
+      GlobalStore.state.grapherSvgPanZoom;
+    if (grapherSvgPanZoom === undefined) {
+      throw new Error("There is no grapher pan zoom instance");
+    }
+    if (enable) {
+      grapherSvgPanZoom.enablePan();
+    } else {
+      grapherSvgPanZoom.disablePan();
+    }
+  }
+
+  onMouseDown(event: MouseEvent): void {
+    BaseTool.updateInvertedCTMMatrix();
+
+    if (event.metaKey) {
+      BaseMoveTool.enablePan(true);
+      return;
+    }
+
+    BaseMoveTool.enablePan(false);
+    this.onMouseDownInternal(event);
+  }
+
+  onMouseUp(event: MouseEvent): void {
+    BaseTool.updateInvertedCTMMatrix();
+    this.onMouseUpInternal(event);
+  }
+
+  onMouseMove(event: MouseEvent): void {
+    BaseTool.updateInvertedCTMMatrix();
+    this.onMouseMoveInternal(event);
+  }
+
+  // override these functions in the extended classes
+  /* eslint-disable @typescript-eslint/no-unused-vars,
+    @typescript-eslint/no-empty-function */
+  onMouseDownInternal(event: MouseEvent): void {}
+
+  onMouseUpInternal(event: MouseEvent): void {}
+
+  onMouseMoveInternal(event: MouseEvent): void {}
+  /* eslint-enable @typescript-eslint/no-unused-vars,
+    @typescript-eslint/no-empty-function */
```

Comment:
Instead of creating these new functions, could we have `ToolSelectMove` call `super.onMouseMove(event)`, etc.?

Reply:
This is called the "Template Pattern": https://en.wikipedia.org/wiki/Template_method_pattern

The intent is that the derived classes implement the "how" to do something, but don't control the "when" to do something.  This way all of the derived classes don't need to remember to call the super function, but instead the base class controls when they have their functions called.

Reply:
Sounds good!